plugins/browser-plugin-media/src/sessionTracking.ts

```diff
@@ -0,0 +1,35 @@
+import { SelfDescribingJson } from '@snowplow/tracker-core';
+import { buildMediaPlayerSessionEntity } from './core';
+import { MediaSessionTrackingStats } from './sessionStats';
+import { MediaPlayerAdBreak, MediaPlayer, MediaPlayerEventType } from './types';
+
+/**
+ * Manages the media player session that is optionally added as a context entity
+ * in media tracking.
+ */
+export class MediaSessionTracking {
+  /// Same as the ID of the media tracking.
+  private id: string;
+  private startedAt: Date;
+  private pingInterval?: number;
+  private stats = new MediaSessionTrackingStats();
+
+  constructor(id: string, startedAt?: Date, pingInterval?: number) {
+    this.id = id;
+    this.pingInterval = pingInterval;
+    this.startedAt = startedAt ?? new Date();
+  }
+
+  update(eventType: MediaPlayerEventType | undefined, mediaPlayer: MediaPlayer, adBreak?: MediaPlayerAdBreak) {
```

Comment:
```suggestion
  update(eventType?: MediaPlayerEventType, mediaPlayer: MediaPlayer, adBreak?: MediaPlayerAdBreak) {
```

Reply:
That would be better, but after merging the suggestion I had to take it back, because TypeScript complained that the first property can't be made optional.